src/hssm/plotting/model_cartoon.py

```diff
@@ -0,0 +1,1985 @@
+    add_legend: bool = True,
+    alpha: float = 1.0,
+    alpha_pp: float = 0.05,
+    alpha_trajectories: float = 0.5,
+    keep_frame: bool = False,
+    random_state: int | None = None,
+    **kwargs,
+):
+    """Calculate and plot posterior predictive for a model.
+
+    Parameters
+    ----------
+    model_name : str
+        Name of the model to simulate.
+    axis : matplotlib.axes.Axes
+        The axis to plot on.
+    theta_mean : pandas.DataFrame, optional
+        Mean parameter values for simulation.
+    theta_posterior : pandas.DataFrame, optional
+        Posterior samples of parameters.
+    data : pandas.DataFrame, optional
+        Observed data to plot.
+    n_trajectories : int, default=10
+        Number of trajectories to plot.
+    bin_size : float, default=0.05
+        Size of bins for histograms.
+    n_samples : int, default=10
+        Number of posterior samples to use.
+    linewidth_histogram : float or int, default=0.5
+        Line width for histogram elements.
+    linewidth_model : float or int, default=0.5
+        Line width for model cartoon elements.
+    legend_fontsize : int, default=7
+        Font size for legend.
+    legend_shadow : bool, default=True
+        Whether to add shadow to legend box.
+    legend_location : str, default="upper right"
+        Location of legend.
+    delta_t_model : float, default=0.01
+        Time step size for model cartoon plotting.
+    add_legend : bool, default=True
+        Whether to add legend to plot.
+    alpha : float, default=1.0
+        Transparency for main plot elements.
+    alpha_pp : float, default=0.05
+        Transparency for posterior predictive samples.
+    alpha_trajectories : float, default=0.5
+        Transparency for trajectory paths.
+    keep_frame : bool, default=False
+        Whether to keep the frame around the plot.
+    random_state : int, optional
+        Random seed for reproducibility.
+    **kwargs
+        Additional keyword arguments passed to plotting functions.
+
+    Notes
+    -----
+    This function visualizes model predictions by plotting simulated trajectories,
+    histograms of response times, and model cartoons. It can show both the mean
+    prediction and uncertainty from posterior samples.
+    """
+    color_dict = {
+        -1: "black",
+        0: "black",
+        1: "green",
+        2: "blue",
+        3: "red",
+        4: "orange",
+        5: "purple",
+        6: "brown",
+    }
+
+    ylim_low, ylim_high = kwargs.get("ylims", (0, 5))
+    xlim_low, xlim_high = kwargs.get("xlims", (0, 5))
+
+    # # Extract some parameters from kwargs
+    axis.set_xlim(xlim_low, xlim_high)
+    axis.set_ylim(ylim_low, ylim_high)
+    bins = list(np.arange(xlim_low, xlim_high, bin_size))
+
+    # ADD MODEL:
+
+    # RUN SIMULATIONS
+    # -------------------------------
+
+    # Simulator Data
+    if random_state is not None:
+        np.random.seed(random_state)
+
+    rand_int = np.random.choice(400000000)
```

Comment:
Separate this constant too?

Reply:
what do you mean here?

Reply:
You can also define a constant for 400000000 here and reuse it across the file